checker/src/main/java/alluxio/checker/HiveIntegrationChecker.java

```diff
@@ -0,0 +1,202 @@
+      if (mUserMode == 1) {
+        useAsSource(con, tableName);
+      } else {
+        useAsUnderFS(con, tableName);
+      }
+
+      sql = "describe " + tableName;
+      try (PreparedStatement describeTablePS = con.prepareStatement(sql)) {
+        describeTablePS.execute();
+      }
+
+      sql = "select * from " + tableName;
+      try (PreparedStatement selectTablePS = con.prepareStatement(sql)) {
+        selectTablePS.execute();
+      }
+    } catch (Exception e) {
+      printExceptionReport(e, reportWriter);
+      return 1;
+    }
+    reportWriter.println("Congratulations, you have configured Hive with Alluxio correctly!\n");
+    reportWriter.println(TEST_PASSED_MESSAGE);
+    return 0;
+  }
+
+  /**
+   * Supports the first integration way: use Alluxio as one option to store Hive tables.
+   *
+   * @param con the Hive connection
+   * @param tableName the name of the test table
+   */
+  private void useAsSource(Connection con, String tableName) throws Exception {
+    String sql = "CREATE TABLE " + tableName + " (ROW1 STRING, ROW2 STRING) "
+        + "ROW FORMAT DELIMITED FIELDS TERMINATED BY '|' "
+        + "LOCATION '" + mAlluxioURL + "/alluxioTestFolder/'";
+    try (PreparedStatement createTablePS = con.prepareStatement(sql)) {
+      createTablePS.execute();
+    }
+  }
+
+  /**
+   * Supports the second integration way: use Alluxio as the default filesystem.
+   *
+   * @param con the Hive connection
+   * @param tableName the name of the test table
+   */
+  private void useAsUnderFS(Connection con, String tableName) throws Exception {
+    String sql = "CREATE TABLE " + tableName + " (ROW1 STRING, ROW2 STRING) "
+        + "ROW FORMAT DELIMITED FIELDS TERMINATED BY '|' "
+        + "STORED AS TEXTFILE";
+    try (PreparedStatement createTablePS = con.prepareStatement(sql)) {
+      createTablePS.execute();
+    }
+
+    sql = "INSERT INTO " + tableName + " VALUES ('Hive', 'Test')";
+    try (PreparedStatement loadTablePS = con.prepareStatement(sql)) {
+      loadTablePS.executeUpdate();
+    }
+  }
+
+  /**
+   * Translates the exception message to user-facing message.
+   *
+   * @param exception the thrown exception when running Hive queries
+   * @param reportWriter save user-facing messages to a generated file
+   */
+  private void printExceptionReport(Exception exception, PrintWriter reportWriter) {
+    String exceptionStr = exception.toString();
+    if (exceptionStr.contains("Class alluxio.hadoop.FileSystem not found")) {
+      reportWriter.println(FAIL_TO_FIND_CLASS_MESSAGE);
+      reportWriter.println(TEST_FAILED_MESSAGE);
+    } else if (exceptionStr.contains("No FileSystem for scheme \"alluxio\"")
+        || exceptionStr.contains("No FileSystem for scheme: alluxio")) {
+      reportWriter.println(FAIL_TO_FIND_FS_MESSAGE);
+      reportWriter.println(TEST_FAILED_MESSAGE);
+    } else {
+      reportWriter.println("Other errors occur, please fix it and "
```

Comment:
"Other errors" doesn't mean anything to the user since they don't know about the if-else structure of this method